agent/src/protocol.ts

```diff
@@ -132,3 +145,12 @@ export interface ExecuteRecipeParams {
     id: RecipeID
     humanChatInput: string
 }
+
+export interface QuickPickParams {
+    items: string[]
```

Comment:
How about making these objects as well? This allows us to add more metadata in the future like icons or detailed text.
```suggestion
    items: QuickPickItem[]
```

Reply:
Something in this direction, which avoids the `default?: string` property
```ts
interface QuickPickItem {
  id: string
  title: string
  default?: boolean
}
```

Reply:
In the long-term, it might make sense to eventually have icons for things like fuzzy symbol finders. It's not something we need to have in the protocol right now, but I think it's nice to be able to add new properties to quickpick items if we want to.

In the short-term, having a separate ID and title is useful.

Reply:
should the item have `default`? I guess it's fine for if we want to select more than one

Reply:
I prefer `QuickPickItem.default: true` over `QuickPickParams.default`